### PR TITLE
docs: update tag docs with interactive tags

### LINKED ISSUE #3449: Interactive tag: Usage and Style tab docs
## Acceptance criteria

- Add content to the Usage and Style tabs on our website for Tag to encompass the new Interactive tag variant and its counterparts (Dismissable, Selectable, Operational). Stephanie Cree (@skcele) will be helping create the doc content. 
- Use the [Usage template](https://carbondesignsystem.com/contributing/documentation/#usage-template:-for-components-with-multiple-variants) and [Style template](https://carbondesignsystem.com/contributing/documentation/#style-template) to help you write the documentation.

---

**To-do**

- [x] Update Usage tab with content
- [x] Update Style tab with content
- [x] Review
- [x] PR

----

Also will close #3818

### PR DESCRIPTION
**Pull request by laurenmrice**
**Closes** #3449 and #3818 

This PR updates the Tag component Usage and Style documentation with our current content templates and additions of Interactive tags (Dismissible, Selectable, and Operational)

---

#### Changelog

**New**

- New sections to match our content templates
- New content and images for the Dismissible, Selectable, and Operational tags.

**Removed**

- Old images

### CODE REVIEW

**Comment by laurenmrice** [comment]
I need help with getting the newest changes of the A11y section addition, change of anchor links, and live demo section reorder to appear on my Tag Usage tab here.

**Comment by alisonjoseph** [comment]
@laurenmrice fixed!

**Review by Kritvi-bhatia17 — COMMENTED** [review]
Hi @laurenmrice, I've added some comments. If you agree with them, you can commit the changes. If not, you can simply ignore them.

Also, I have a small question/suggestion: Do you think naming the images by adding numerical prefixes will be helpful?

File: src/pages/components/tag/style.mdx [hunk 1]
```diff
@@ -1,75 +1,214 @@
...
 ## Structure
 
-All tags have the same height. However, the width of tags varies based on the
-amount of content. All four corners of a tag are rounded with a 24px radius.
+There are three fixed heights of tags—large (32px), medium (24px), and small
+(18px). However, the width of tags can vary based on the length of the title.
 
 | Element  | Property                    | px / rem   | Spacing token |
 | -------- | --------------------------- | ---------- | ------------- |
+| Tag (lg) | height                      | 32 / 2     | –             |
```
[the remark this answers is not in the document]
Comment (laurenmrice) [reply src/pages/components/tag/style.mdx:208]: I am going to change all sizes for tag on the usage and style tabs to start from small to large in the content and images to resolve the inconsistency.

**Review by guidari — APPROVED** [review]
LGTM! 🚀

**Review by Kritvi-bhatia17 — APPROVED** [review]
Amazing work Lauren!! 🔥

File: src/pages/components/tag/images/tag-style-structure-operational-examples.png
Comment (thyhmdo) [inline src/pages/components/tag/images/tag-style-structure-operational-examples.png:1]: This image looks very small compared to others. It is probably the fixed size image issue. If you think it looks ok, I'm fine with that.

File: src/pages/components/tag/usage.mdx [hunk 2]
```diff
@@ -40,35 +58,395 @@ them.
...
+
+For further content guidance, see Carbon’s
+[content guidelines](https://carbondesignsystem.com/guidelines/content/overview/).
+
+### States
+
+The four tag variants have different states:
+
+- **Read-only tags:** enabled, disabled, and skeleton
+- **Dismissible and operational tags:** enabled, hover, focus, on click,
```
[the remark this answers is not in the document]
Comment (laurenmrice) [reply src/pages/components/tag/usage.mdx:219]: Oh yes, very true. Thanks Thy!

File: src/pages/components/tag/images/tag-style-structure-operational-examples.png
Comment (laurenmrice) [reply src/pages/components/tag/images/tag-style-structure-operational-examples.png:1]: Yes, this is wrong. I will fix it!

**Review by thyhmdo — APPROVED** [review]
Looks great!

**Pull request merged**